programs/jito-protecc/src/lib.rs

```diff
@@ -15,23 +16,54 @@ pub mod jito_protecc {
         Ok(())
     }
 
-    pub fn pre_guard(ctx: Context<PreGuard>, bump: u8) -> Result<()> {
+    pub fn pre_guard(
+        ctx: Context<PreGuard>,
+        // If true, then performs a negative balance change check on the guarded account's lamports.
+        should_guard_lamports: bool,
+        bump: u8,
+    ) -> Result<()> {
         let guarded_state = &mut ctx.accounts.guarded_state;
-        guarded_state.lamports = ctx.accounts.guarded_account.lamports();
+
+        if should_guard_lamports {
+            guarded_state.maybe_pre_lamports = Some(ctx.accounts.guarded_account.lamports());
```

Comment:
does it hurt to always do this? we have storage space anyway, maybe makes more sense as arg to second post-guard

Reply:
Reason I like doing it this way is so client doesn't have to try and fetch lamports balance from an RPC which wont be guaranteed to be accurate if they're blasting txs. I kept it optional since mevoors might actually incur a negative balance change on specific strategies and they're fine with that

Reply:
when would this happen? signer isn't a token balance

Reply:
guarded_account is any arbitrary account the mevoor wants to protect. The account can be the same as the fee payer here or in their mev txs, and can also be the tipper on bundles. But if it's not any of those but IS a signer of some tx, then they may want to make sure their lamports don't change negatively